### PR TITLE
feat: `OTPInput` component redesign

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
## Description
OTPInput Redesign 

## Changes

- Add size prop
- Update font family & size being used
- Update visuals

## Component Checklist

- [ ] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [ ] Add Interaction Tests (if applicable)
- [ ] Add changeset

### CODE REVIEW

**Comment by anuraghazra** [comment]
Why is the mask square shaped? 
image

expected design?

File: packages/blade/src/components/Input/BaseInput/BaseInput.tsx [hunk 1]
```diff
@@ -309,6 +309,11 @@ type BaseInputCommonProps = FormInputLabelProps &
      * ```
      */
     trailingButton?: React.ReactElement<LinkProps>;
+    /**
+     * Whether to use Text or Heading component for Input text
+     * @default text
+     **/
+    valueComponentType?: 'text' | 'heading';
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/BaseInput/BaseInput.tsx:316]: Don't have much context on this, what exactly was the usecase for this?

File: packages/blade/src/components/Input/OTPInput/OTPInput.stories.tsx [hunk 2]
```diff
@@ -181,7 +202,12 @@ export default {
 } as Meta<OTPInputProps>;
 
 const OTPInputTemplate: StoryFn<typeof OTPInputComponent> = ({ ...args }) => {
-  return <OTPInputComponent {...args} />;
+  const maxWidth = args.otpLength === 4 ? '376px' : '568px';
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/OTPInput/OTPInput.stories.tsx:205]: So the OTPInput itself doesn't have MaxWidth is it?

File: packages/blade/src/components/Typography/Heading/getHeadingStyles.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const getHeadingStyles = ({
+  weight,
+  size,
+  theme,
+  color,
+}: Pick<HeadingProps, 'weight' | 'size' | 'color'> & {
+  theme: Theme;
+}): CSSObject => {
+  return getBaseTextStyles({ ...getHeadingProps({ weight, size, color }), theme });
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Heading/getHeadingStyles.ts:15]: nice

**Review by anuraghazra — APPROVED** [review]

File: packages/blade/src/components/Input/OTPInput/OTPInput.stories.tsx (hunk 2 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/OTPInput.stories.tsx:205]: Can't find it in ts so guessing nahi hai. We can add it if someone raises it though

File: packages/blade/src/components/Input/BaseInput/BaseInput.tsx (hunk 1 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/BaseInput/BaseInput.tsx:316]: So for displaying value of the input, we sometimes need it to have the styles of our "Text" component and other times styles of our "Heading" component. Text & Heading differ in sizes as well as font families. OTPInput needs the input value to be shown as the same styles as Heading component whereas other input components need the same style as Text component

**Pull request merged**